src/main/java/mate/academy/rickandmorty/controller/CharacterController.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.rickandmorty.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.internal.CharacterDto;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Characters", description = "Work with Rick and Morty characters")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/characters")
+public class CharacterController {
+    private final CharacterService characterService;
+
+    @Operation(summary = "Get a random character",
+            description = "Returns a random character from the database.")
+    @GetMapping
+    public CharacterDto getRandomCharacter() {
+        return characterService.getRandomCharacter();
+    }
+
+    @Operation(summary = "Search characters by name",
+            description = "Returns a list of characters matching the given name.")
+    @GetMapping("/search")
```

Comment:
```suggestion
    @GetMapping("/random")
    public CharacterDto getRandomCharacter() {
        return characterService.getRandomCharacter();
    }

    @Operation(summary = "Search characters by name",
            description = "Returns a list of characters matching the given name.")
    @GetMapping
```
it will be more RESTful (we generally avoid verbs in URL paths, HTTP verb should imply action)